cmd/controlloop/controlloop.go

```diff
@@ -125,9 +126,19 @@ func handleSignals(stopChannel chan struct{}, signals ...os.Signal) {
 }
 
 func newPodController(stopChannel chan struct{}) (*controlloop.PodController, error) {
-	cfg, err := rest.InClusterConfig()
+	var cfg *rest.Config
+	var err error
+	cfg, err = rest.InClusterConfig()
```

Comment:
If this is run inside the cluster, but the KUBECONFIG env variable is set:

* ip.go will use NewReconcileLooperWithKubeconfig due to the if condition
* newPodController will use rest.InClusterConfig because in cluster rest.InClusterConfig will succeed

So your code can end up using the in cluster config for one, and the kubeconfig for the other